includes/legacy/class-wc-legacy-coupon.php

```diff
@@ -109,7 +109,7 @@ public function __get( $key ) {
 				$value = $this->get_usage_count();
 			break;
 			case 'expiry_date' :
-				$value = $this->get_date_expires();
+				$value = ( $this->get_date_expires() ? $this->get_date_expires()->getOffsetTimestamp() : '' );
```

Comment:
I think this may be incorrect - in the data store you save this as Y-m-d. Shouldn't this be the same return value? Did 2.6.x use Y-m-d or timestamp?